test/commands/tenant.js

```diff
@@ -0,0 +1,139 @@
+    getTenants: () => Promise.resolve(TENANTS),
+    setTenant: () => Promise.resolve(TENANTS),
+    removeTenant: () => Promise.resolve(TENANTS)
+  }
+};
+
+test.beforeEach(t => {
+  t.context.log = console.log;
+});
+
+test.afterEach(t => {
+  console.log = t.context.log;
+});
+
+test.serial.cb('tenantCommand() should resolve to a message displaying previous and current tenants', (t) => {
+  console.log = function (content) {
+    t.is(content, `Current Tenant: ${TENANTS.current || 'Unset'}
+Previous Tenants:
+  ${TENANTS.previous.sort().join(`
+  `)}`);
+  };
+
+  tenantCommand(INPUT, FLAGS, OPTIONS)
+    .then(() => t.end())
+    .catch(error => {
+      t.fail(error);
+      t.end();
+    });
+});
+
+test.serial.cb('tenantCommand() should call blinkMobileIdentity.getTenants() if no tenant is passed in', (t) => {
+  t.plan(1);
+  const options = {
+    blinkMobileIdentity: {
+      getTenants: () => {
+        t.pass();
+        return Promise.resolve(TENANTS);
+      }
+    }
+  };
+  console.log = function () {};
+
+  tenantCommand([], FLAGS, options)
+    .then(() => t.end())
+    .catch(error => {
+      t.fail(error);
+      t.end();
+    });
+});
+
+test.serial.cb('tenantCommand() should call blinkMobileIdentity.setTenant() if a tenant is passed in with no --remove flag', (t) => {
+  t.plan(1);
+  const options = {
+    blinkMobileIdentity: {
+      setTenant: (tenant) => {
+        t.is(tenant, TENANT_NAME);
+        return Promise.resolve(TENANTS);
+      }
+    }
+  };
+  console.log = function () {};
+
+  tenantCommand(INPUT, FLAGS, options)
+    .then(() => t.end())
+    .catch(error => {
+      t.fail(error);
+      t.end();
+    });
+});
+
+test.serial.cb('tenantCommand() should call blinkMobileIdentity.removeTenant() if a tenant is passed in with a --remove flag', (t) => {
+  t.plan(1);
+  const options = {
+    blinkMobileIdentity: {
+      removeTenant: (tenant) => {
+        t.is(tenant, TENANT_NAME);
+        return Promise.resolve(TENANTS);
+      }
+    }
+  };
+  console.log = function () {};
+
+  tenantCommand(INPUT, { remove: true }, options)
+    .then(() => t.end())
+    .catch(error => {
+      t.fail(error);
+      t.end();
+    });
+});
+
+test.serial.cb('tenantCommand() should log error if any tenant function rejects with error', (t) => {
+  t.plan(3);
+  const options = {
+    blinkMobileIdentity: {
+      getTenants: () => Promise.reject('Errror Message'),
+      setTenant: () => Promise.reject('Errror Message'),
+      removeTenant: () => Promise.reject('Errror Message')
+    }
+  };
+  console.log = function (content) {
+    t.is(content, `
+There was a problem attempting to use the tenant command:
+
+Errror Message
+
+Please fix the error and try again.
+`);
+  };
+
+  Promise.all([
+    tenantCommand([], FLAGS, options),
+    tenantCommand(INPUT, FLAGS, options),
+    tenantCommand(INPUT, { remove: true }, options)
+  ])
+    .then(() => t.end())
+    .catch(error => {
+      t.fail(error);
+      t.end();
```

Comment:
If you don't use `.cb()`, then ava will allow the test function to return a `Promise`. Then it will automatically wait for it to resolve or reject, and will automatically record a test failure if it rejects. Doing it this way could significantly reduce boilerplate from your test files.